models/concerns/validate_data.rb

```diff
@@ -128,6 +128,12 @@ def check_services(schedule)
   def check_relation_consistent_timewindows
     unconsistent_relation_timewindows = []
     @hash[:relations].each{ |relation|
+      if Models::Relation::ON_VEHICLES_TYPES.include?(relation[:type]) && relation[:linked_ids].to_a.any?
+        raise OptimizerWrapper::DiscordantProblemError.new("#{relation[:type]} does not expect any service ids")
+      elsif Models::Relation::ON_SERVICES_TYPES.include?(relation[:type]) && relation[:linked_vehicle_ids].to_a.any?
+        raise OptimizerWrapper::DiscordantProblemError.new("#{relation[:type]} does not expect any vehicle ids")
+      end
```

Comment:
in fact we can do this more cleanly with `case .. when`
```suggestion
      case relation[:type]
      when *Models::Relation::ON_VEHICLES_TYPES
        if relation[:linked_ids].to_a.any? || relation[:linked_vehicle_ids].to_a.empty? 
          raise OptimizerWrapper::DiscordantProblemError.new("#{relation[:type]} relations do not support linked_ids and expect linked_vehicle_ids")
        end
      when *Models::Relation::ON_SERVICES_TYPES
         # opposite of the above
      when  *Models::Relation::ON_NOTHING_TYPES # only if necessary 
         # then we raise if there is any linked_ids or linked_vehicle_ids
      else
        raise 'Unknown relation type' # in case we ever forget to handle a new relation type
      end
```